controller/kubernetes_pv_controller_test.go

```diff
@@ -61,53 +61,57 @@ func (tc *KubernetesTestCase) copyCurrentToExpect() {
 	tc.expectVolume = tc.volume.DeepCopy()
 }
 
-func newPV() *apiv1.PersistentVolume {
-	pvcFilesystemMode := apiv1.PersistentVolumeFilesystem
-	return &apiv1.PersistentVolume{
+func newPV() *corev1.PersistentVolume {
+	pvcFilesystemMode := corev1.PersistentVolumeFilesystem
+	return &corev1.PersistentVolume{
 		ObjectMeta: metav1.ObjectMeta{
 			Name: TestPVName,
 		},
-		Spec: apiv1.PersistentVolumeSpec{
-			Capacity: apiv1.ResourceList{
-				apiv1.ResourceStorage: *resource.NewQuantity(1, resource.BinarySI),
+		Spec: corev1.PersistentVolumeSpec{
+			Capacity: corev1.ResourceList{
+				corev1.ResourceStorage: *resource.NewQuantity(1, resource.BinarySI),
 			},
 			VolumeMode: &pvcFilesystemMode,
-			PersistentVolumeSource: apiv1.PersistentVolumeSource{
-				CSI: &apiv1.CSIPersistentVolumeSource{
-					Driver: types.LonghornDriverName,
-					FSType: "ext4",
-					VolumeAttributes: map[string]string{
-						"numberOfReplicas":    "3",
-						"staleReplicaTimeout": "30",
-					},
-					VolumeHandle: TestVolumeName,
-				},
+			PersistentVolumeSource: corev1.PersistentVolumeSource{
+				CSI: newPVSourceCSI(),
 			},
-			ClaimRef: &apiv1.ObjectReference{
+			ClaimRef: &corev1.ObjectReference{
 				Name:      TestPVCName,
 				Namespace: TestNamespace,
 			},
 			StorageClassName: TestStorageClassName,
 		},
-		Status: apiv1.PersistentVolumeStatus{
-			Phase: apiv1.VolumeBound,
+		Status: corev1.PersistentVolumeStatus{
+			Phase: corev1.VolumeBound,
+		},
+	}
+}
+
+func newPVSourceCSI() *corev1.CSIPersistentVolumeSource {
```

Comment:
Actually you can add this as `newPVSourceCSI`, but return `corev1.PersistentVolumeSource` instead to save repeated the outside structure `PersistentVolumeSource`.